src/vm/runners/cairo_runner.zig

```diff
@@ -155,6 +161,15 @@ pub const CairoRunner = struct {
         self.run_ended = true;
     }
 
+    /// Ensures that the trace is relocated and is retrievable from the VM and returns it.
+    pub fn consolidateTrace(self: *Self) ![]RelocatedTraceEntry {
+        const relocation_table = try self.vm.segments.relocateSegments(self.allocator);
+        try self.vm.relocateTrace(relocation_table);
+
+        const relocated_trace = self.vm.trace_context.state.enabled.relocated_trace_entries.items;
```

Comment:
Would be good to use the `vm.getRelocatedTrace` method, here to encapsulate details.